llama_index/evaluation/dataset_generation.py

```diff
@@ -131,21 +138,39 @@ def _node_question_generator(
                 use_async=True,
             )
             response = query_engine.query(
-                self.question_gen_query,
+                self.qna_gen_query if qna_mode else self.question_gen_query,
             )
 
             result = str(response).strip().split("\n")
-            cleaned_questions = [
+            cleaned_output = [
                 re.sub(r"^\d+[\).\s]", "", question).strip() for question in result
             ]
-            questions.extend(cleaned_questions)
+            output_strings.extend(cleaned_output)
 
-        questions = [question for question in questions if question != ""]
+        output_strings = [string for string in output_strings if string != ""]
 
         if num is not None:
-            questions = questions[:num]
-        return questions
+            output_strings = output_strings[:num]
+        return output_strings
 
     def generate_questions_from_nodes(self, num: Optional[int] = None) -> List[str]:
         """Generates questions for each document."""
-        return self._node_question_generator(self.nodes, num)
```

Comment:
maybe we can keep the old name, but use a boolean toggle like `generate_questions_from_nodes(include_answers=True)`